### PR TITLE
Add APIs for CEIPOptIn configuration

### LINKED ISSUE #43: Add API for CEIP Opt-In configuration
**Describe the feature request**
To support CEIP , CLI should prompt and store the users CEIP Opt-in choice in the configuration. So APIs are needed to set/get the CEIP OptIn configuration 
**Describe alternative(s) you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by prkalle**
### What this PR does / why we need it
This PR add API for CEIP Opt-in configuration. 
The new config parameter is added in the tanzu config-ng.yaml as shown below

```
cli:
    ceipOptIn: "false"
```

### Which issue(s) this PR fixes

Fixes #43 

### Describe testing done for PR
Unit tests and CI passed

### Release note

```release-note
Add API for CEIP Opt-In configuration
```

### CODE REVIEW

File: config/types/clientconfig_types.go [hunk 1]
```diff
@@ -301,6 +303,10 @@ type ClientConfig struct {
 
 	// ClientOptions are client specific options like feature flags, environment variables, repositories, discoverySources, etc.
 	ClientOptions *ClientOptions `json:"clientOptions,omitempty" yaml:"clientOptions,omitempty"`
+
+	// ClientOptionsNG are nextgen client specific options like feature flags, environment variables, ceipOptIn, etc
+	// that goes into nextgen configuration file.
+	ClientOptionsNG *ClientOptions `json:"clientOptionsNG,omitempty" yaml:"clientOptionsNG,omitempty"`
```
[the remark this answers is not in the document]
Comment (prkalle) [reply config/types/clientconfig_types.go:309]: Thanks @mpanchajanya. After our discussion, updated the PR with option 1).

File: config/types/clientconfig_types.go [hunk 2]
```diff
@@ -175,6 +175,8 @@ type CLIOptions struct {
 	//
 	// Deprecated: CompatibilityFilePath has been deprecated and will be removed from future version
 	CompatibilityFilePath string `json:"compatibilityFilePath,omitempty" yaml:"compatibilityFilePath,omitempty"`
+	// CEIPOptIn is the users CEIP opt-in/opt-out status.
+	CEIPOptIn string `json:"ceipOptIn,omitempty" yaml:"ceipOptIn,omitempty"`
```
Comment (mpanchajanya) [inline config/types/clientconfig_types.go:179]: @prkalle Could you remove it from here

Comment (prkalle) [reply config/types/clientconfig_types.go]: Updated. Thanks for catching this!

**Review by mpanchajanya — APPROVED** [review]

**Review by anujc25 — APPROVED** [review]
LGTM

**Pull request merged**